enterprise/server/remote_execution/runner/runner_linux.go

```diff
@@ -0,0 +1,37 @@
+//go:build linux && !android
+
+package runner
+
+import (
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/container"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/bare"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/docker"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/firecracker"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/podman"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/sandbox"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/platform"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+)
+
+func (p *pool) initContainerProviders() {
+	providers := make(map[platform.ContainerType]container.Provider)
+	dockerProvider, err := docker.NewProvider(p.env, p.imageCacheAuth, p.hostBuildRoot())
+	if err != nil {
+		log.Warningf("Failed to initialize docker container provider: %s", err)
+	}
```

Comment:
This should return an error that propagates up and causes the executor to fail initialization. Otherwise the executor could be in a state where it appears healthy but is unable to run tasks.

(Same for all the log.Warnings elsewhere)

Reply:
Makes sense. Fixed!